src/commands/redis_cmd.cc

```diff
@@ -2385,19 +2393,74 @@ class CommandZAdd : public Commander {
 
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
     int ret;
+    double old_score = member_scores_[0].score;
+    bool incr = (flags_ & kZSetIncr) != 0;
```

Comment:
This line can be safely moved before the actual use of `incr` - before `if`-statement.